openslides_backend/action/actions/motion/set_state.py

```diff
@@ -61,3 +64,33 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
         )
         instance["last_modified"] = round(time.time())
         return instance
+
+    def check_permissions(self, instance: Dict[str, Any]) -> None:
+        motion = self.datastore.get(
+            FullQualifiedId(Collection("motion"), instance["id"]),
+            [
+                "state_id",
+                "submitter_ids",
+                "meeting_id",
+            ],
+        )
+        if has_perm(
+            self.datastore,
+            self.user_id,
+            Permissions.Motion.CAN_MANAGE_METADATA,
+            motion["meeting_id"],
+        ):
+            return
+
+        state = self.datastore.get(
+            FullQualifiedId(Collection("motion_state"), motion["state_id"]),
+            ["allow_submitter_edit"],
+        )
+        if state.get("allow_submitter_edit") and self.is_user_submitter(
+            motion["submitter_ids"]
+        ):
+            return
```

Comment:
I would suggest to move the state check also into the mixin and rename the method, since in all instances where you call `is_user_submitter` you also check the state each time, so this should be combined.

Reply:
Move the get state/check allow_submitter_edit code into the mixin. Update the tests.